src/lavatory/utils/artifactory.py

```diff
@@ -112,51 +103,63 @@ def filter(self, repo, terms=[], depth=3, sort=None, offset=None, limit=None):
         the default n items will be enough.
         """
 
-        terms.append({"repo": {"$eq": repo}})
+        if terms is None:
+            terms = []
+
+        terms.append({"repo": {"$eq": self.repo_name}})
         terms.append({"type": {"$eq": "folder"}})
         terms.append({"depth": {"$eq": depth}})
 
-        findexpr = json.dumps({"$and": terms})
-    
-        aql = "items.find({})".format(findexpr)
-    
+        find_expr = json.dumps({"$and": terms})
+
+        aql = "items.find({})".format(find_expr)
+
         if sort:
             aql += ".sort({})".format(json.dumps(sort))
-    
+
         if offset:
             aql += ".offset({})".format(offset)
-    
+
         if limit:
             aql += ".limit({})".format(limit)
-    
+
         LOG.debug("AQL: {}".format(aql))
-    
-        response = self.http.post("/search/aql", aql)
 
-        return response["results"]
-    
-    def retain(self, repo, specproject, depth, terms=None, count=None, weeks=None):
+        response = self.artifactory.find_by_aql(criteria=aql)
+
+        results = response['results']
+
+        return results
+
+    def retain(self, spec_project, depth, terms=None, count=None, weeks=None):
         if [terms, count, weeks].count(None) != 2:
             raise ValueError("Must specify exactly one of terms, count, or weeks")
 
-        if weeks:
-            now = datetime.datetime.now()
-            before = now - datetime.timedelta(weeks=weeks)
-            created = before.strftime("%Y-%m-%dT%H:%M:%SZ")
-
         purgable = []
 
-        for project in self.filter(repo, depth=depth):
-            if specproject and specproject!=project["name"]:
+        for project in self.filter(self.repo_name, depth=depth):
```

Comment:
Dont need to pass in `self.repo_name` since it is already part of the class

Reply:
ah missed that one. removed.